classes/class-wc-connect-nux.php

```diff
@@ -83,5 +94,168 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			);
 			return $pointers;
 		}
+
+		public function get_jetpack_install_status() {
+			$status = Jetpack_Install_Status::UNINSTALLED;
+
+			if( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+				$status = Jetpack_Install_Status::INSTALLED;
+			}
+
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				// installed, but not activated
+				return $status;
+			}
+
+			if ( defined( 'JETPACK_DEV_DEBUG' ) ) {
+				$status = Jetpack_Install_Status::DEV;
+				return $status;
+			}
+
+			if ( ! method_exists( 'Jetpack_Data', 'get_access_token' )
+				|| ! class_exists( 'Jetpack' )
+			) {
+				$status = Jetpack_Install_Status::ERROR;
+				return $status;
+			}
+			$status = Jetpack_Install_Status::ACTIVATED;
+
+
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			if ( $user_token
+				&& is_object( $user_token )
+				&& isset( $user_token->external_user_id )
+			) {
+				$status = Jetpack_Install_Status::CONNECTED;
+			};
+
+			return $status;
+		}
+
+		public function show_relevant_nux_notices() {
+			$jetpack_install_status = $this->get_jetpack_install_status();
+			switch ( $jetpack_install_status ) {
+				case Jetpack_Install_Status::ACTIVATED:
+					add_action( 'admin_notices', array( $this, 'show_banner_before_connection_welcome' ) );
+					add_action( 'admin_notices', array( $this, 'show_banner_before_connection_get_access' ) );
+					break;
+				case Jetpack_Install_Status::CONNECTED:
+					add_action(
+						'admin_notices',
+						array( $this, 'show_banner_after_connection_see_how_it_works' )
+					);
+					break;
+			}
+		}
+
+		public function show_banner_before_connection_get_access( $content ) {
+			if( function_exists( 'get_current_screen' ) ) {
```

Comment:
After looking at the Codex and inspecting the hook firing order (using the [Debug Bar Actions and Filters Addon](https://wordpress.org/plugins/debug-bar-actions-and-filters-addon/)), I think we're good to remove this check.

The codex states:
> The function returns null if called from the `admin_init` hook. It should be OK to use in a later hook such as `current_screen`.

I'm seeing `admin_notices` occur after `current_screen`.

For your consideration, an alternate approach could be to hook into `current_screen` action initially, which gets the `$screen` object passed to it, determine which notices belong on that screen, and _then_ attach the appropriate display callbacks to the `admin_notices` action.